crates/storage/db/src/implementation/mdbx/tx.rs

```diff
@@ -13,7 +13,20 @@ use crate::{
 use parking_lot::RwLock;
 use reth_interfaces::db::{DatabaseWriteError, DatabaseWriteOperation};
 use reth_libmdbx::{ffi::DBI, Transaction, TransactionKind, WriteFlags, RW};
-use std::{marker::PhantomData, str::FromStr, sync::Arc, time::Instant};
+use reth_tracing::tracing::debug;
+use std::{
+    backtrace::Backtrace,
+    marker::PhantomData,
+    str::FromStr,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    time::{Duration, Instant},
+};
+
+/// Duration after which we emit the log about long-lived database transactions.
+const LONG_TRANSACTION_DURATION: Duration = Duration::from_secs(30);
```

Comment:
I think we can do this, if the tx has been open for that long, the perf hit of backtrace capture is negligible imo